### ISSUE TITLE
RVM segment doesn't parse for -head version of ruby

### ISSUE THREAD

**Issue by rjorgenson** [bhilburn/powerlevel9k#575]
With the recent fix to the RVM segment it now only prints the last column in the awk separation, for the -head version of ruby(and possibly others) this just results in the version being reported as 'head' which isn't terribly useful =] How does the below look? This should split out the path reported and print just the ruby version which we can then parse out as we wish. The below just does what the previous segment did, but we could expand on it if we want to display more/less information?

```shell
echo $MY_RUBY_HOME | awk -F'/' '{print $NF}' | awk -F'-' '{print $2}'
```

**Comment by bhilburn** [comment]
@rjorgenson - Ah, good catch. Agreed, just printing `HEAD` is really not useful at all for version managers, hah. Yes, let's get back to actual version strings.

Your suggestion looks clean & simple. If there is a way to do it with ZSH regex (paging @docwhat?), that would be great. Otherwise, can you file the fix against `master`?

**Comment by rjorgenson** [comment]
I didn't know ZSH could do anything like that. Got a link to any documentation on how it works? I will definitely look into that as I thought double awk calls was pretty clunky =]

**Comment by docwhat** [comment]
Had to do some lookups; I don't use `rvm` much anymore (though I've contributed to it a bunch in the past).

So it looks like weird `awk`s are being used instead of just asking `rvm`.  `rvm` is really fast because it is written in pure shell functions; so using that would be better.

Let me double check things out in a docker container.

**Comment by rjorgenson** [comment]
Submitted PR #582 before I saw your comment @docwhat - The segment now does it in shell and doesn't invoke the overhead of RVM by calling it. The awk call was from years ago, RVM was pretty slow and calling it from the prompt(at least in bash, where I first wrote this prompt code) was problematic. Thoughts on the solution I came up with?

**Comment by docwhat** [comment]
Okay, what a pain... I ran into rvm/rvm#3742 while trying to install `ruby-head`. Since I'm me, I had to track it down.

So how about this?

``` zsh
prompt_rvm() {
  local version_and_gemset=${rvm_env_string/ruby-}

  if [[ -n "$version_and_gemset" ]]; then
    "$1_prompt_segment" "$0" "$2" "240" "$DEFAULT_COLOR" "$version_and_gemset" 'RUBY_ICON'
  fi
}
```

`rvm_env_string` is almost exactly what was there before: the ruby version and the gemset.

- If the gemset is set it is appended with an `@` sign.
- If you do `rvm use system` then it is empty. You cannot create gemsets for `system`.

If the version starts with `ruby-` then it is stripped off.  Otherwise the whole thing is shown.

e.g. "jruby-9.1.7.0" is shown in its entirety.

This is in PR #583
